src/common/config_proxy.h

```diff
@@ -124,11 +124,16 @@ class ConfigProxy {
   ConfigValues* operator->() noexcept {
     return &values;
   }
-#ifdef WITH_SEASTAR
+  ConfigValues get_config_values() const {
+    std::lock_guard l{lock};
+    return values;
+  }
   void set_config_values(const ConfigValues& val) {
+#ifndef WITH_SEASTAR
```

Comment:
Just wondering, do we really need this `#ifndef WITH_SEASTAR`? Wouldn't the lock work for seastar too?

Reply:
They try to avoid all locking in seastar -- I think that's why there are those 100% unsafe `operator->` overloads.